docs/custom_component.md

```diff
@@ -39,31 +39,35 @@ The easiest way to implement a `TransformComponent` is to subclass the provided
 chunks as pandas dataframes.
 
 ```python
+import pandas as pd
 from fondant.component import PandasTransformComponent
+from fondant.executor import PandasTransformExecutor
+
 
 class ExampleComponent(PandasTransformComponent):
-    
-    def setup(self, argument1, argument2):
-        """This method is called once per component with any custom arguments it received. Use 
-        it for component wide setup or to store your arguments as instance attributes to access 
-        them in the `transform` method.
-        
+
+    def __init__(self, *args, argument1, argument2) -> None:
+        """
         Args:
-            argumentX: A custom argument passed to the component
-        """ 
+            argumentX: An argument passed to the component
+        """
+        # Initialize your component here based on the arguments
 
-    def transform(self, dataframe):
-        """Implement your custom transformation logic in this single method
-        
+    def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
+        """Implement your custom logic in this single method
         Args:
-            dataframe: A Pandas dataframe containing one partition of your data
-            
+            dataframe: A Pandas dataframe containing the data
```

Comment:
why omit that it's one partition? might make things a bit more clear to the user if they want to go more in-depth

Reply:
This wasn't a conscious change, will revert.